### PR TITLE
Attributed code snippets

### LINKED ISSUE #97: Snippets: file name and attribution
Snippets now have file names, but they don't yet have attribution. Also, the file names aren't actually picked up in the UI.

_If_ we use third-party snippets that aren't public domain (for example, for sample code, it oftentimes doesn't matter), I think we should extend `ICodeSnippet` (perhaps add an `IThirdPartyCodeSnippet` that inherits from it) where we can properly attribute the code.

Once a snippet has been solved, it should then show both the file name and, if applicable, the attribution.

### PR DESCRIPTION
**Pull request by chucker**
fixes #97

### CODE REVIEW

**Review by ricardoboss — APPROVED** [review]
Maybe we should link to the archive.org version of the websites instead, but this is a good base to work with for future snippets

File: Prolangle/Pages/Snippets.razor [hunk 1]
```diff
@@ -44,14 +44,45 @@
...
+			}
+
+			@if (attributedSnippet.License is not { Length: > 0 } license)
+			{
+				<span>.</span>
+			}
+			else
+			{
+				<span>, licensed as follows: <strong style="font-weight: 500">@license</strong>.</span>
+			}
```
[the remark this answers is not in the document]
Comment (chucker) [reply Prolangle/Pages/Snippets.razor:74]: Hmm. Do we want to split those interfaces and let a snippet implement two different ones? Then I would make the properties non-nullable.

[the remark this answers is not in the document]
Comment (ricardoboss) [reply Prolangle/Pages/Snippets.razor:74]: (Also see my other comment)

Yes, I think that would be a good design

File: Prolangle/Snippets/IAttributedCodeSnippet.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	/// The attribution for this code snippet. Will be revealed when the user
+	/// has won.
+	/// </summary>
+	string Attribution { get; }
+
+	/// <summary>
+	/// The license for this code snippet. If set, will be revealed when the
+	/// user has won.
+	/// </summary>
+	string? License { get; }
```
[the remark this answers is not in the document]
Comment (chucker) [reply Prolangle/Snippets/IAttributedCodeSnippet.cs]: Yep.

File: Prolangle/Pages/Snippets.razor (hunk 1 above)
[the remark this answers is not in the document]
Comment (chucker) [reply Prolangle/Pages/Snippets.razor]: Done!

**Comment by chucker** [comment]
>Maybe we should link to the archive.org version of the websites instead

Yeah, perhaps. Leaving it as is for now.

**Pull request merged**